server/themes/default/public/javascripts/angular-textreplace.js

```diff
@@ -5,57 +5,58 @@ angular.module('angular-highlight', []).directive('highlight', function() {
 			attrs.highlightClass = 'angular-highlight';
 		}
 		
-		function arrSearch(nameKey, myArray){
-		    for (var i=0; i < myArray.length; i++) {
-		    	var rx = new RegExp(myArray[i].match, "gi");
-		        if (nameKey.search(rx) > -1) {
-		            return myArray[i];
-		        }
-		    }
-		}
+		//scope.highlight has the original message
+		//scope.replacement has all the replacements
+		
+		//scope.replacement[i].highlight shows the mode
+		//scope.replacement[i].match contains the match
+		//scope.replacement[i].replace contains the replacement
 		
-		var rReplacer = function(match, item) {
-			var resultObject = arrSearch(match, scope.replacement);
+		function rexReplacer(inputHtml) {
+			//Store the original HTML for later use
+			var html = inputHtml;
 			
-			//Fix issue with undefined variables, mostly from migrated configs.
-			if (typeof resultObject.highlight !== 'undefined') {
-				var thisMode = resultObject.highlight;
-			} else {
-				var thisMode = false;
-			}
+			//Perform this function for every Text/Regex Match
+			scope.replacement.forEach(function(textreplace) {
+			
+				//Fix issue with undefined variables, mostly from migrated configs.
+				if (typeof textreplace.highlight !== 'undefined') {
```

Comment:
Could be replaced with
```
const thisMode = textreplace.highlight || false;
```

`var` is outdated, `let` and `const` should be used for numerous reasons. If/else-declarations should be replaced with `let x; if (true) x = a; else x = b;`, with ternary operators or with the syntax shown above.